app/controllers/uploads_controller.rb

```diff
@@ -20,7 +20,7 @@ def create
       failed = load_csv.failed_instances
       @upload.check_for_headers
 
-      validation_warnings = failed.map(&:display_errors_with_row).sort
+      validation_warnings = failed.sort_by { |x| x.errors.messages[:row][0].to_i }.map(&:display_errors_with_row)
```

Comment:
Sort by `:row` attribute.  ActiveRecord model validations (except for `Program`) do not populate the `:row` attribute and render first as `Row N/A`